lib/openai/assistant.rb

```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+
+require_relative "assistant/version"
+require_relative "assistant_obj"
+require "json"
+require "net/http"
+require "rest-client"
+module Openai
+  # An openai assistant
+  class Assistant
+    @openai_api_key = nil
+    @openai_url = nil
+
+    # @param api_key [String] The api key of openai
+    def initialize(api_key = "")
+      @openai_api_key = api_key
+      # hard the host because if the official docs change the host, maybe it will change another
+      # we need to update this gem for any change
+      @openai_url = "https://api.openai.com/v1/assistants"
+    end
```

Comment:
I think you can split the  @openai_api_key and @base_openai_url into a Base class such as `class Openai::Base`, and Assistant class will inherit from that class such as `class Openai::Assistant < Base`

Reply:
updated!